### PR TITLE
Fix hidden range embedded in UAX29URLEmail grammar

### LINKED ISSUE #12561: UAX29URLEmailTokenizerImpl.jflex matches emails with commas and invalid periods in the local part
### Description

The `UAX29URLEmailTokenizerImpl.jflex` code matches commas as part of email local part, as well as invalid leading, trailing, or consecutive periods. Examples of bad matches: `foo,bar@yahoo.com`, `,bar@yahoo.com`, `bar.@yahoo.com`, `.bar@gmail.com`, `foo...bar@gmail.com`. 

https://github.com/apache/lucene/blame/main/lucene/analysis/common/src/java/org/apache/lucene/analysis/email/UAX29URLEmailTokenizerImpl.jflex#L188

Relevant code:
```
EMAILatomText = [A-Za-z0-9!#$%&'*+-/=?\^_`{|}~]
EMAILlabel = {EMAILatomText}+ | {EMAILquotedString}
EMAILlocalPart = {EMAILlabel} ("." {EMAILlabel})*
```

The problem is `EMAILatomText` matches periods and commas, and it was not meant to. Commas are invalid in the local part unless it's quoted (`EMAILquotedString` supports that). Periods are supposed to only be valid as `EMAILlocalPart` allows, but with the bug, the `("." {EMAILlabel})*` part never matches as periods are matched in `EMAILlabel`.

As for why it matches commas and periods? 

See the ASCII table:

image

And look for a hidden range that matches these 5 characters. It's the `+-/` above. Clearly copy-pasted from someplace like this:
image

The dash is a range here, and coincidentally (or is it?) is the middle character in this range as well! (perhaps mitigating discovering this bug)

The solution is as simple as this escaping the dash:
```
EMAILatomText = [A-Za-z0-9!#$%&'*+\-/=?\^_`{|}~]
```
You can of course also just move it to the end or the beginning of the character class where it is taken literally, but seems a bit more fragile, and escaping keeps the "minus" after the "plus" as in the original order this was copied from.

### Version and environment details

Current and all older versions I believe.

### PR DESCRIPTION
**Pull request by rmuir**
TLDR: The dash needed escaping!

See #12561 for an explanation of the issue. Actually there were TODOs already in existing tests. 

Closes #12561

### CODE REVIEW

File: lucene/analysis/common/src/test/org/apache/lucene/analysis/email/TestUAX29URLEmailAnalyzer.java [hunk 1]
```diff
@@ -433,9 +433,9 @@ public void testMailtoSchemeEmails() throws Exception {
         new String[] {
           "mailto",
           "persona@example.com",
-          // TODO: recognize ',' address delimiter. Also, see examples of ';' delimiter use at:
+          // Also, see examples of ';' delimiter use at:
```
Comment (eraneverlaw) [inline lucene/analysis/common/src/test/org/apache/lucene/analysis/email/TestUAX29URLEmailAnalyzer.java:436]: Not sure what this comment means? Semicolon works as a delimiter.

Also, interesting that `?cc=personC@example.com` is supposedly a legal email address.

Comment (rmuir) [reply lucene/analysis/common/src/test/org/apache/lucene/analysis/email/TestUAX29URLEmailAnalyzer.java:436]: yeah i don't know, I just tried to preserve these comments, and there are other similar TODO in the test. Especially this one: https://github.com/apache/lucene/blob/53ba27a63be6849d5383b8bfc6d1508dd7b66f0c/lucene/analysis/common/src/test/org/apache/lucene/analysis/email/TestUAX29URLEmailAnalyzer.java#L428C8-L428C96

**Comment by ztravis** [comment]
It would be great to fix this bug!
